### PR TITLE
Add incremental timeouts to blocksync fetching

### PR DESCRIPTION
**Pull request by Kubuxu**

### CODE REVIEW

File: chain/blocksync/blocksync.go [hunk 1]
```diff
@@ -55,8 +48,8 @@ func ParseBSOptions(optfield uint64) *BSOptions {
 }
 
 const (
-	BSOptBlocks   = 1 << 0
-	BSOptMessages = 1 << 1
+	BSOptBlocks = 1 << iota
```
Comment (whyrusleeping) [inline chain/blocksync/blocksync.go]: this breaks the protocol, no?

File: lib/increadtimeout/incrt.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+func (crt *incrt) Read(buf []byte) (n int, err error) {
+	start := now()
+	err = crt.rd.SetReadDeadline(start.Add(crt.wait))
+	if err != nil {
+		return 0, err
+	}
+
+	defer func() {
+		crt.rd.SetReadDeadline(time.Time{})
```
Comment (whyrusleeping) [inline lib/increadtimeout/incrt.go]: do we care to set this error?

File: chain/blocksync/blocksync.go (hunk 1 above)
Comment (Kubuxu) [reply chain/blocksync/blocksync.go]: It doesn't, it is just a nicer way to write this. The values should be the same.

File: lib/increadtimeout/incrt.go (hunk 2 above)
Comment (Kubuxu) [reply lib/increadtimeout/incrt.go]: I would say no, this removes deadline, if something goes wrong there there isn't anything we can do but we might already have data so...

**Comment by Kubuxu** [comment]
TODO fix `TestSyncSimple`
```
failed to fetch tipset from peer during hello: set pipe: deadline not supported
github.com/filecoin-project/lotus/node/hello.(*Service).HandleStream
	/home/circleci/project/node/hello/hello.go:97
github.com/libp2p/go-libp2p/p2p/host/basic.(*BasicHost).SetStreamHandler.func1
	/go/pkg/mod/github.com/libp2p/go-libp2p@v0.4.2/p2p/host/basic/basic_host.go:393
```

File: chain/blocksync/blocksync.go (hunk 1 above)
Comment (Kubuxu) [reply chain/blocksync/blocksync.go]: Confirmed, values are the same.

Comment (Kubuxu) [reply chain/blocksync/blocksync.go]: Confirmed, values are the same.

**Comment by Kubuxu** [comment]
Tests are fixed.

**Pull request merged**